### PR TITLE
Fixes and Improvements (Null checks, Typescript support, Eslint, Prettier)

### PR DESCRIPTION
**Pull request by stefkudla**
## What's new?
- Added respective null checks to dynamic data when needed
  - Blogs without author data, cover images, etc, will omit their respective components (eg. no cover image in the CMS no cover image rendered on the page)
- Updated astro package and plugins
- Installed proper typescript, eslint, and prettier plugins to work with Astro
- Data coming in from the CMS is now fully Typesafe in the code (better DXE)

## Note
The bucket itself had zero changes

### CODE REVIEW

**Comment by stefkudla** [comment]
Here is the updated Vercel Deployment link: https://vercel.com/new/clone?env=PUBLIC_COSMIC_BUCKET_SLUG,PUBLIC_COSMIC_READ_KEY&repository-url=https://github.com/cosmicjs/simple-astro-blog/tree/api-v3

You can test out how its working on this branch before we merge with this link:

https://vercel.com/new/clone?env=PUBLIC_COSMIC_BUCKET_SLUG,PUBLIC_COSMIC_READ_KEY&repository-url=https://github.com/cosmicjs/simple-astro-blog/tree/fixes-improvements

**Review by tonyspiro — APPROVED** [review]
LGTM!

**Comment by tonyspiro** [comment]
One thing: logo in footer should be light mode version.

File: src/pages/blog/[slug].astro [hunk 1]
```diff
@@ -1,101 +1,125 @@
...
-  });
+      props: { post }
+    }
+  })
 }
 
-const { post } = Astro.props;
+const { post } = Astro.props
+
+console.log(post.metadata.author)
```
Comment (tonyspiro) [inline src/pages/blog/[slug].astro:55]: @stefkudla can remove this too

Comment (stefkudla) [reply src/pages/blog/[slug].astro]: Nice catch

**Review by tonyspiro — APPROVED** [review]
LGTM!

**Pull request merged**